interpreters/quickjs/quickjs.patch

```diff
@@ -0,0 +1,302 @@
+diff --color -upr quickjs-20200705/libregexp.c quickjs-20200705-modified/libregexp.c
+--- quickjs-20200705/libregexp.c	2020-07-25 05:25:41.000000000 +0800
++++ quickjs-20200705-modified/libregexp.c	2020-08-27 16:04:41.440000000 +0800
+@@ -21,6 +21,7 @@
+  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+  * THE SOFTWARE.
+  */
++#include <alloca.h>
+ #include <stdlib.h>
+ #include <stdio.h>
+ #include <stdarg.h>
+diff --color -upr quickjs-20200705/qjs.c quickjs-20200705-modified/qjs.c
+--- quickjs-20200705/qjs.c	2020-07-25 05:25:41.000000000 +0800
++++ quickjs-20200705-modified/qjs.c	2020-08-27 16:23:55.750000000 +0800
+@@ -34,7 +34,7 @@
+ #include <time.h>
+ #if defined(__APPLE__)
+ #include <malloc/malloc.h>
+-#elif defined(__linux__)
++#elif defined(__linux__) || defined(__NuttX__)
```

Comment:
Huang Qi, I think many of these modifications here could be submitted to QuickJS project. What do you think? For now I think we can merge it this way, but it could be on external project mainline to improve NuttX support.

Reply:
Sure, I'll try to submit this patch to quickjs later.

Reply:
Thank you! If everything is fine, please remove the Draft tag.